modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/AppendProcessorTests.java

```diff
@@ -198,14 +198,21 @@ public void testAppendingToListWithDuplicatesDisallowed() throws Exception {
         String originalField = RandomDocumentPicks.addRandomField(random(), ingestDocument, list);
         List<String> expectedValues = new ArrayList<>(list);
         List<String> existingValues = randomSubsetOf(list);
-        int uniqueValuesSize = randomIntBetween(0, 10);
-        List<String> uniqueValues = new ArrayList<>();
-        for (int i = 0; i < uniqueValuesSize; i++) {
-            uniqueValues.add(randomAlphaOfLengthBetween(1, 10));
+        int nonexistingValuesSize = randomIntBetween(0, 10);
+        List<String> nonexistingValues = new ArrayList<>();
+        for (int i = 0; i < nonexistingValuesSize; i++) {
+            boolean valueExists = true;
+            while (valueExists) {
```

Comment:
This seems really confusing to me, I think we could do the same thing with:

```java
int nonexistingValueSize = randomIntBetween(0, 10);
// generate n new values
Set<String> newValues = Stream.generate(() -> randomAlphaOfLengthBetween(1, 10)).limit(nonexistingValueSize).collect(Collectors.toSet());
// create a set using the new values, making sure there are no overlapping values already present in the existing values
Set<String> nonexistingValues = Sets.difference(newValues, new HashSet<>(existingValues));
```

Using the while loop for this is a bit overkill, I think `Set`s makes it easier to avoid duplicates

Reply:
Thanks, @dakrone. I simplified the test as you suggested.